src/hypothesis/database.py

```diff
@@ -240,7 +240,11 @@ def fetch(self, key):
     def save(self, key, value):
         path = self._value_path(key, value)
         if not os.path.exists(path):
-            tmpname = path + '.' + str(binascii.hexlify(os.urandom(16)))
+            suffix = binascii.hexlify(os.urandom(16))
+            if not isinstance(suffix, str):
```

Comment:
python2 `str` is python3 `bytes`.
python3 `str` is python2 `unicode`. 

So on python2 the `str` function returns a bytes object and on python3 it returns a unicode object. Does `binascii.hexlify` return differently based on version as well? Then maybe it doesn't matter.